arch/generic/crc32_braid_c.c

```diff
@@ -211,6 +207,1279 @@ Z_INTERNAL uint32_t PREFIX(crc32_braid)(uint32_t crc, const uint8_t *buf, size_t
+#endif
+        in5 ^= next5 ^ a1;
+        in6 ^= next6 ^ a2 ^ b1;
+        in7 ^= next7 ^ a3 ^ b2 ^ c1;
+        in8 ^= next8 ^ a4 ^ b3 ^ c2 ^ d1;
+
+        e1 = (in5 << 17);
+        e2 = (in5 >> 15) ^ (in5 << 23);
+        e3 = (in5 >> 9) ^ (in5 << 19);
+        e4 = (in5 >> 13);
+        e6 = (in5 << 12);
+        e7 = (in5 >> 20);
+
+        f1 = (in6 << 17);
+        f2 = (in6 >> 15) ^ (in6 << 23);
+        f3 = (in6 >> 9) ^ (in6 << 19);
+        f4 = (in6 >> 13);
+        f6 = (in6 << 12);
+        f7 = (in6 >> 20);
+
+        g1 = (in7 << 17);
+        g2 = (in7 >> 15) ^ (in7 << 23);
+        g3 = (in7 >> 9) ^ (in7 << 19);
+        g4 = (in7 >> 13);
+        g6 = (in7 << 12);
+        g7 = (in7 >> 20);
+
+        h1 = (in8 << 17);
+        h2 = (in8 >> 15) ^ (in8 << 23);
+        h3 = (in8 >> 9) ^ (in8 << 19);
+        h4 = (in8 >> 13);
+        h6 = (in8 << 12);
+        h7 = (in8 >> 20);
+
+        in9 = input[i/sizeof(uint32_t) + 8];
+        in10 = input[i/sizeof(uint32_t) + 9];
+#if BYTE_ORDER == BIG_ENDIAN
+        in9 = ZSWAP32(in9);
+        in10 = ZSWAP32(in10);
+#endif
+        in9 ^= next9 ^ b4 ^ c3 ^ d2 ^ e1;
+        in10 ^= next10 ^ a6 ^ c4 ^ d3 ^ e2 ^ f1;
+
+        i1 = (in9 << 17);
+        i2 = (in9 >> 15) ^ (in9 << 23);
+        i3 = (in9 >> 9) ^ (in9 << 19);
+        i4 = (in9 >> 13);
+        i6 = (in9 << 12);
+        i7 = (in9 >> 20);
+
+        j1 = (in10 << 17);
+        j2 = (in10 >> 15) ^ (in10 << 23);
+        j3 = (in10 >> 9) ^ (in10 << 19);
+        j4 = (in10 >> 13);
+        j6 = (in10 << 12);
+        j7 = (in10 >> 20);
+
+        out1 = a7 ^ b6 ^ d4 ^ e3 ^ f2 ^ g1;
+        out2 = b7 ^ c6 ^ e4 ^ f3 ^ g2 ^ h1;
+        out3 = c7 ^ d6 ^ f4 ^ g3 ^ h2 ^ i1;
+        out4 = d7 ^ e6 ^ g4 ^ h3 ^ i2 ^ j1;
+        out5 = e7 ^ f6 ^ h4 ^ i3 ^ j2;
+        out6 = f7 ^ g6 ^ i4 ^ j3;
+        out7 = g7 ^ h6 ^ j4;
+        out8 = h7 ^ i6;
+        out9 = i7 ^ j6;
+        out10 = j7;
+
+        next1 = out1;
+        next2 = out2;
+        next3 = out3;
+        next4 = out4;
+        next5 = out5;
+        next6 = out6;
+        next7 = out7;
+        next8 = out8;
+        next9 = out9;
+        next10 = out10;
+
+    }
+#if BYTE_ORDER == BIG_ENDIAN
+    next1 = ZSWAP32(next1);
+    next2 = ZSWAP32(next2);
+    next3 = ZSWAP32(next3);
+    next4 = ZSWAP32(next4);
+    next5 = ZSWAP32(next5);
+    next6 = ZSWAP32(next6);
+    next7 = ZSWAP32(next7);
+    next8 = ZSWAP32(next8);
+    next9 = ZSWAP32(next9);
+    next10 = ZSWAP32(next10);
+#endif
+
+    memcpy(final, input+(i/sizeof(uint32_t)), len-i);
+    final[0] ^= next1;
+    final[1] ^= next2;
+    final[2] ^= next3;
+    final[3] ^= next4;
+    final[4] ^= next5;
+    final[5] ^= next6;
+    final[6] ^= next7;
+    final[7] ^= next8;
+    final[8] ^= next9;
+    final[9] ^= next10;
+
+    crc = crc32_braid_base(crc, (uint8_t*) final, len-i);
+
+    return crc;
+}
+
+Z_INTERNAL uint32_t PREFIX(crc32_braid)(uint32_t crc, const uint8_t *buf, size_t len) {
```

Comment:
I propose:
* `crc32_braid` should be renamed `crc32_c` and moved into `arch\generic\crc32_c.c`
* `crc32_braid_base` should be renamed back to `crc32_braid`.
* `chorba_*` functions should be moved to `arch\generic\crc32_chorba_c.c`
  * Possibly all `chorba_*` functions prefixed with `crc32_`
@Dead2 thoughts?